src/main/java/com/ms/zookeeper/authz/ZkFileAuthorizer.java

```diff
@@ -0,0 +1,139 @@
+package com.ms.zookeeper.authz;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.log4j.Logger;
+
+class FileData {
+    private long timestamp;
+    private Set<String> users;
+
+    long getTimestamp() {
+        return timestamp;
+    }
+    void setTimestamp(final long timestamp) {
+        this.timestamp = timestamp;
+    }
+    Set<String> getUsers() {
+        return users;
+    }
+    void setUsers(final Set<String> users) {
+        this.users = users;
+    }
+};
+
+public class ZkFileAuthorizer implements ZkAuthorizer, Runnable {
+
+    private static final Logger LOG = Logger.getLogger(ZkFileAuthorizer.class);
+
+    private static volatile ZkFileAuthorizer instance = null;
+
+    //expr and set of users who are allowed
```

Comment:
what is proper style? I suspect // expr ... (with space after //. i am sure there is some sort of java style checker.

Reply:
Yes, space after `//`; that is for single line comment, and `/** */` for multi line comment.

Reply:
fixed